src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -518,11 +517,9 @@ private void MakeConstantTuple(LocalSymbol inProgress, BoundExpression boundInit
                     var diagnostics = DiagnosticBag.GetInstance();
                     Debug.Assert(inProgress != this);
                     var type = this.Type;
-                    if (boundInitValue == null)
-                    {
-                        var inProgressBinder = new LocalInProgressBinder(this, this._initializerBinder);
-                        boundInitValue = inProgressBinder.BindVariableOrAutoPropInitializer(_initializer, this.RefKind, type, diagnostics);
-                    }
+
+                    var inProgressBinder = new LocalInProgressBinder(this, this._initializerBinder);
+                    BoundExpression boundInitValue = inProgressBinder.BindVariableOrAutoPropInitializer(_initializer, this.RefKind, type, diagnostics);
```

Comment:
@cston As far as I have been able to figure out, the only difference is that with code [like this](https://github.com/svick/roslyn/blob/a6e9529/src/Compilers/CSharp/Test/Semantic/Semantics/ColorColorTests.cs#L2118-L2130), `const Color Color = Color.Red;` produces the same warning twice, while `const Color color = Color.Red;` produces the warning only once.

Though my solution means that the warning is always produced twice (and then deduped), which is not better, so I'm going to revert this change.